### PR TITLE
passing block as namespace

### PR DESCRIPTION
**Pull request by lokanadhamm**

### CODE REVIEW

**Comment by lokanadhamm** [comment]
Will be usefull for Multitenant applications to avoid multiple namespaced clients

File: lib/redis/namespace.rb [hunk 1]
```diff
@@ -252,6 +254,14 @@ def warning?
       @warning
     end
 
+    def namespace_is_block?
+      @namespace_is_block
+    end
+
+    def namespace_string
+      namespace_is_block? ? @namespace.call : @namespace
```
Comment (rafaelfranca) [inline lib/redis/namespace.rb]: Can we avoid this method call and just use the instance variable here?

File: lib/redis/namespace.rb [hunk 2]
```diff
@@ -234,6 +234,8 @@ class Namespace
 
     def initialize(namespace, options = {})
       @namespace = namespace
+      #respond_to? is costly. So, will assing the result to @namespace_is_block and use that variable
```
Comment (rafaelfranca) [inline lib/redis/namespace.rb]: ✂️ this comment.

File: README.md [hunk 3]
```diff
@@ -29,6 +29,10 @@ namespaced_redis.get('foo')
 # => nil
 redis_connection.get('ns:foo')
 # => nil
+
+#Redis::Namespace will also support Proc as a namespace and will take the resultant string as namespace at run time.
```
Comment (rafaelfranca) [inline README.md]:
```suggestion
Redis::Namespace also supports `Proc` as a namespace and will take the result string as namespace at runtime.
```

File: README.md [hunk 4]
```diff
@@ -29,6 +29,10 @@ namespaced_redis.get('foo')
...
+namespaced_redis = Redis::Namespace.new(Proc.new {Tenant.current_tenant}, :redis => redis_connection)
```
Comment (rafaelfranca) [inline README.md]: Need to wrap this in a codeblock

**Comment by iloveitaly** [comment]
@lokanadhamm quick reminder on this PR! If you can correct the review comments, I'd love to get this merged.

**Pull request closed without merging**